apps/wallet-dashboard/app/(protected)/staking/page.tsx

```diff
@@ -171,16 +194,31 @@ function StakingDashboardPage(): React.JSX.Element {
                                 </div>
                             </div>
                         </div>
-                        <StakeDialog
-                            stakedDetails={selectedStake}
-                            onSuccess={handleOnStakeSuccess}
-                            isOpen={isDialogStakeOpen}
-                            handleClose={handleCloseStakeDialog}
-                            view={stakeDialogView}
-                            setView={setStakeDialogView}
-                            selectedValidator={selectedValidator}
-                            setSelectedValidator={setSelectedValidator}
-                        />
+                        {isDialogStakeOpen && (
+                            <StakeDialog
+                                stakedDetails={selectedStake}
+                                onSuccess={handleOnStakeSuccess}
+                                handleClose={handleCloseStakeDialog}
+                                view={stakeDialogView}
+                                setView={setStakeDialogView}
+                                selectedValidator={selectedValidator}
+                                setSelectedValidator={setSelectedValidator}
+                                onUnstakeClick={handleUnstakeClick}
+                            />
+                        )}
+
+                        {isUnstakeDialogOpen && selectedStake && (
+                            <UnstakeDialog
+                                handleClose={() => setIsUnstakeDialogOpen(false)}
+                                extendedStake={selectedStake}
+                                onBack={handleOnUnstakeBack}
+                                view={UnstakeDialogView.Unstake}
+                                onSuccess={() => {
+                                    refetchDelegatedStakes();
```

Comment:
we could wait for tx digest like in onSuccess for StakingDialog. From docs I can see that having tx digest is not a guarantee that it will be available from the apis.